site/source/docs/porting/simd.rst

```diff
@@ -1015,3 +1015,301 @@ The following table highlights the availability and expected performance of diff
      - 💣 emulated with complex SIMD+scalar sequence
 
 Only the 128-bit wide instructions from AVX instruction set are available. 256-bit wide AVX instructions are not provided.
+
+
+====================================================== 
+Compiling SIMD code targeting ARM NEON instruction set
+======================================================
+
+Emscripten supports compiling existing codebases that use ARM NEON by
+passing the `-mfpu=neon` directive to the compiler, and including the
+header `<arm_neon.h>`.
+
+In terms of performance, it is very important to note that only
+instructions which operate on 128-bit wide vectors are supported
+cleanly. This means that nearly any instruction which is not of a "q"
+variant (i.e. "vaddq" as opposed to "vadd") will be scalarized.
+
+These are pulled from `SIMDe repository on Github
+<https://github.com/simd-everywhere/simde>`_. To update emscripten
+with the latest SIMDe version, run `tools/simde_update.py`.
+
+The following table highlights the availability of various 128-bit
+wide intrinsics.
+
+Similarly to above, the following legend is used:
+ - ✅ Wasm SIMD has a native opcode that matches the NEON instruction, should yield native performance
+ - 💡 while the Wasm SIMD spec does not provide a proper performance guarantee, given a suitably smart enough compiler and a runtime VM path, this intrinsic should be able to generate the identical native NEON instruction.
+ - ⚠️ the underlying NEON instruction is not available, but it is emulated via at most few other Wasm SIMD instructions, causing a small penalty.
+ - ❌ the underlying NEON instruction is not exposed by the Wasm SIMD specification, so it must be emulated via a slow path, e.g. a sequence of several slower SIMD instructions, or a scalar implementation.
+ - ⚫ the given NEON intrinsic is not available. Referencing the intrinsic will cause a compiler error.
+
+For detailed information on each intrinsic function, refer to `NEON Intrinsics Reference
+<https://developer.arm.com/architectures/instruction-sets/simd-isas/neon/intrinsics>`_.
+
+.. list-table:: NEON Intrinsics
+   :widths: 20 30
+   :header-rows: 1
+
+   * - Intrinsic name
+     - Wasm SIMD Support
+   * - vaba
+     - ⚫ Not implemented, will trigger compiler error
+   * - vabal
+     - ⚫ Not implemented, will trigger compiler error
+   * - vabd
+     - ⚫ Not implemented, will trigger compiler error
+   * - vabdl
+     - ⚫ Not implemented, will trigger compiler error
+   * - vabs
+     - native
```

Comment:
Great work on this! This looks like a spot on companion to the x86 SSE functions.

The data in this table is a good start, but it looks too coarse to be accurate? Like the doc mentioned above, the 64-bit wide functions are all scalarized, so e.g. it is not correct that `vabs` here would be native. For example, looking in detail at the absolute value functions, here is how I'd present it:
```
  * - vabs_f32, vabs_s8, vabs_s16, vabs_s32
     - 💣 64-bit wide SIMD operations are scalarized
  * - vabs_f64, vabs_s64
     - ✅ implemented as native fabs and llabs functions
  * - vabsq_f32, vabsq_s8, vabsq_s16 and vabsq_s32
     - ✅ wasm_f32x4_abs, wasm_i8x16_abs, wasm_i16x8_abs and wasm_i32x4_abs
  * - vabsq_f64, vabsq_s64
     - ❌ scalarized
```

I would recommend presenting the data in a form where each intrinsic function name is referenced in full, so that one can ctrl-f search for the exact name of any intrinsic function. To avoid repetitive tedium with multiple functions having the same diagnosis, bucket the names on a single line like `vabs_f64, vabs_s64` above.

(btw implementation of `vabs_f64` and `vabs_s64` look a bit sloppy, they equate to regular double `abs`, but the implementation has loops and structs and ternaries in it, leaving it to the optimizer to do magic, which it might not want to do if one runs in optimize for size mode or debug mode)